microcosm_flask/url.py

```diff
@@ -0,0 +1,58 @@
+"""
+Url fatories.
+
+"""
+
+from microcosm_flask.operations import Operation, NODE_PATTERN
+
+
+def url_extractor_factory(ns=None, operation=None, use_model_id=None, identifier_key=None, **url_string_args):
+    """
+    A factory that creates url extractors.
+    url extractor is an (component, model) -> string function that can generates url for relevant resources
+    The extractors can component.ns and component.identifier_key to easily generates relevant urls.
+
+    Simple examples:
+
+    class Controller:
+        def __init__(self, graph):
+            self.ns = Namespace(subject="company")
+            self.identifier_key = "company_id"
+    controller = Controller()
+    company = Company(id="ID", clock=3)
+
+    url_extractor_factory()(ctrl, company) -> '/api/v1/company/ID'
+    url_extractor_factory(min_clock=lambda cont, obj: obj.clock)(ctrl, company) -> '/api/v1/company/ID?min_clock=3'
+
+    :ns              - Model namespace. component.ns will get used instead if set to None.
+    :operation       - microcosm_flask.operations.Operation, default is Operation.Retrieve
+    :url_string_args - Allow to specify query args to use
+                       Passed as a dictionary of {arg_name: lambda component, model: action}
+                       For example: {
+                           company_event_id: lambda component, company_event: company_event.id
+                           min_clock: lambda component, company_event: company_event.clock
+                        }
+    :use_model_id    - Should use model.id to create the url.
```

Comment:
The identifier handling feels like it's missing an abstraction.

Reply:
Agree. I've put it inside the extractor to get access to `component.identifier_key` - but I can separate it if I would allow to generate the keys dynamically